### PR TITLE
Get_User_By_Id_Endpoint

### LINKED ISSUE #11: set up getUserById endpoint
set up user/:id/get endpoint to retrieve a certain user's data

### PR DESCRIPTION
**Pull request by ctc-devops**
set up user/:id/get endpoint to retrieve a certain user's data
closes #11

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -13,3 +14,13 @@ app.use(
 );
 
 app.listen(PORT, () => {});
+
+app.get('/users/:id', async (req, res) => {
+  try {
+    const { userId } = req.params;
+    const user = await pool.query('SELECT * FROM users WHERE user_id = $1', [userId]);
+    res.json(user.rows[0]);
```
Comment (henryg0) [inline app.js:22]: add res.status(200) if success, res.status(400) if failure

File: app.js [hunk 2]
```diff
@@ -13,3 +14,13 @@ app.use(
 );
 
 app.listen(PORT, () => {});
+
+app.get('/users/:id', async (req, res) => {
```
Comment (henryg0) [inline app.js:18]: merge w. dev, add code to routes/users/users.js

File: routes/users/users.js [hunk 3]
```diff
@@ -5,6 +5,19 @@ const pool = require('../../db');
 const userRouter = express();
 
 userRouter.use(express.json());
+
+// Get user by id endpoint
+userRouter.get('/users/:id', async (req, res) => {
```
Comment (r-shar) [inline routes/users/users.js]: update url to '/:id'

File: routes/users/users.js [hunk 4]
```diff
@@ -5,6 +5,19 @@ const pool = require('../../db');
...
+  try {
+    const { userId } = req.params;
```
Comment (r-shar) [inline routes/users/users.js]: the param variable name (currently 'userId') _must_ match the param name used in the url ('id')

File: routes/users/users.js [hunk 5]
```diff
@@ -5,6 +5,19 @@ const pool = require('../../db');
...
+    const user = await pool.query('SELECT * FROM users WHERE user_id = $1', [userId]);
```
Comment (r-shar) [inline routes/users/users.js]: - update param variable name according the previous comment
- the column name for user id in the users table is 'id' in the database (update query to reflect this)

File: routes/users/users.js [hunk 6]
```diff
@@ -5,6 +5,19 @@ const pool = require('../../db');
...
+    res.json(user.rows[0]);
```
Comment (r-shar) [inline routes/users/users.js]: res.status(200).json(user.rows)

File: routes/users/users.js [hunk 7]
```diff
@@ -5,6 +5,19 @@ const pool = require('../../db');
...
+  } catch (err) {
+    res.json();
```
Comment (r-shar) [inline routes/users/users.js]: res.status(500).json(err.message)

**Pull request merged**